File: Leetcode1886. Determine whether matrix can be obtained through rotation/Solution.cs
public class Solution {
    public bool FindRotation(int[][] mat, int[][] target) {
        int n = mat.Length;
        // at most 4 rotations
        for (int k = 0; k < 4; k++) {
            // rotation operation
            for (int i = 0; i < n / 2; i++) {
                for (int j = 0; j < (n + 1) / 2; j++) {
                    int temp = mat[i][j];
                    mat[i][j] = mat[n - 1 - j][i];
                    mat[n - 1 - j][i] = mat[n - 1 - i][n - 1 - j];
                    mat[n - 1 - i][n - 1 - j] = mat[j][n - 1 - i];
                    mat[j][n - 1 - i] = temp;
                }
            }

            if (IsEqual(mat, target)) {
                return true;
            }
        }
        return false;
    }

    private bool IsEqual(int[][] mat, int[][] target) {
        int n = mat.Length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] != target[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Leetcode1886. Determine whether matrix can be obtained through rotation/Solution.rs
impl Solution {
    pub fn find_rotation(mat: Vec<Vec<i32>>, target: Vec<Vec<i32>>) -> bool {
        let n = mat.len();
        let mut mat = mat;
        
        // at most 4 rotations
        for _ in 0..4 {
            // rotation operation
            for i in 0..n/2 {
                for j in 0..(n+1)/2 {
                    let temp = mat[i][j];
                    mat[i][j] = mat[n-1-j][i];
                    mat[n-1-j][i] = mat[n-1-i][n-1-j];
                    mat[n-1-i][n-1-j] = mat[j][n-1-i];
                    mat[j][n-1-i] = temp;
                }
            }
            
            if mat == target {
                return true;
            }
        }
        false
    }
}
